fem/pfespace.cpp

```diff
@@ -1842,9 +1842,13 @@ int ParFiniteElementSpace::PackDof(int entity, int index, int edof) const
 static int bisect(const int* array, int size, int value)
 {
    const int* end = array + size;
-   const int* pos = std::lower_bound(array, end, value);
-   MFEM_VERIFY(pos != end, "value not found");
-   return pos - array;
+   const int* pos = std::upper_bound(array, end, value);
+   MFEM_VERIFY(pos != array, "value not found");
+   if (pos == end)
+   {
+      MFEM_VERIFY(*(array+size - 1) == value, "Last entry must be exact")
+   }
+   return pos - array - 1;
```

Comment:
The impact of this can be best illustrated by considering a sequence of P2 triangles and quads. p2 triangles have no face dofs, so {Q, T, T, Q, Q} would gives {0, 1, 1, 1, 2} as the dof numbering. Searching with lower bound for 1, would give the first triangle (T), whereas the true target is the second quad (Q). This was the scenario that allowed me to catch this bug, but this fix appears to have resolved a lot of other failing cases I was working on. My hypothesis for this was that this error was causing the NC mesh to get into a mess such that the indexing no longer worked for any other dofs either.